### PR TITLE
[`unnecessary_cast`] Do not lint negative hexadecimal literals when cast as floats

### PR DESCRIPTION
**Pull request by kraktus**
fix https://github.com/rust-lang/rust-clippy/issues/9603

changelog: [`unnecessary_cast`] Do not lint negative hexadecimal literals when cast as floats

### CODE REVIEW

File: clippy_lints/src/casts/unnecessary_cast.rs [hunk 1]
```diff
@@ -59,9 +59,6 @@ pub(super) fn check<'tcx>(
                 lint_unnecessary_cast(cx, expr, literal_str, cast_from, cast_to);
                 return false;
             },
-            LitKind::Int(_, LitIntType::Unsuffixed) | LitKind::Float(_, LitFloatType::Unsuffixed) => {
-                return false;
-            },
```
Comment (giraffate) [inline clippy_lints/src/casts/unnecessary_cast.rs:64]: Why are these removed? It looks like that the fix for `clippy_utils/src/numeric_literal.rs` is sufficient to me.

Comment (kraktus) [reply clippy_lints/src/casts/unnecessary_cast.rs:64]: It was dead code in the first place

Comment (kraktus) [reply clippy_lints/src/casts/unnecessary_cast.rs:64]: To be exact not dead but if the branching is not taken, it will still result to the function returning `false` and since the code triggered afterward is cheap it’s not worth the extra complexity of an early return

**Comment by giraffate** [comment]
@bors r+

Thanks!
